core/src/main/java/com/linecorp/armeria/client/ClientFactoryBuilder.java

```diff
@@ -560,6 +561,8 @@ private ClientFactoryOptions buildOptions() {
      * Returns a newly-created {@link ClientFactory} based on the properties of this builder.
      */
     public ClientFactory build() {
+        // To initialize the context storage when the factory is built not the first request is sent.
```

Comment:
Asserts are often disabled, it's better to make it more explicit I think - can add an `init` method and call it in a static initializer instead.

Reply:
So you mean adding `RequestContextUtil.init()` and call it from the static initializer of `ClientFactory`? If `init()` doesn't do anything, isn't it disabled as well?

Reply:
Ah, you meant making `contextStorage` not final and initializing in `init()`.

Reply:
Ah nah I meant an empty method - it would still trigger classload and is a but clearer at the call site `init()` as a verb rather than just `storage()`